tests/test_standalone_stac_item.py

```diff
@@ -21,35 +20,11 @@ def quote_none_safe(url):
 
 @pytest.mark.online
 def test_standalone_stac_item_thredds_ncml():
-    thredds_url = "https://pavics.ouranos.ca/twitcher/ows/proxy/thredds"
-    thredds_path = "birdhouse/testdata/xclim/cmip6"
-    thredds_nc = "sic_SImon_CCCma-CanESM5_ssp245_r13i1p2f1_2020.nc"
-    thredds_catalog = f"{thredds_url}/catalog/{thredds_path}/catalog.html"
-    thredds_ds = f"{thredds_path}/{thredds_nc}"
-    thredds_ncml_url = (
-        f"{thredds_url}/ncml/{thredds_path}/{thredds_nc}"
-        f"?catalog={quote_none_safe(thredds_catalog)}&dataset={quote_none_safe(thredds_ds)}"
-    )
-
-    # FIXME: avoid hackish workarounds
-    data = requests.get(thredds_ncml_url).text
-    attrs = xncml.Dataset.from_text(data).to_cf_dict()
-    attrs["access_urls"] = {  # FIXME: all following should be automatically added, but they are not!
-        "HTTPServer": f"{thredds_url}/fileServer/{thredds_path}/{thredds_nc}",
-        "OPENDAP": f"{thredds_url}/dodsC/{thredds_path}/{thredds_nc}",
-        "WCS": f"{thredds_url}/wcs/{thredds_path}/{thredds_nc}?service=WCS&version=1.0.0&request=GetCapabilities",
-        "WMS": f"{thredds_url}/wms/{thredds_path}/{thredds_nc}?service=WMS&version=1.3.0&request=GetCapabilities",
-        "NetcdfSubset": f"{thredds_url}/ncss/{thredds_path}/{thredds_nc}/dataset.html",
-    }
-
+    url = "https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/birdhouse/testdata/xclim/cmip6/sic_SImon_CCCma-CanESM5_ssp245_r13i1p2f1_2020.nc"
+    attrs = ncattrs(url)
     stac_item_id = CMIP6populator.make_cmip6_item_id(attrs["attributes"])
     stac_item = STAC_item_from_metadata(stac_item_id, attrs, CMIP6ItemProperties, GeoJSONPolygon)
-
-    ref_file = os.path.join(CUR_DIR, "data/stac_item_testdata_xclim_cmip6_ncml.json")
-    with open(ref_file, mode="r", encoding="utf-8") as ff:
-        reference = json.load(ff)
-
-    assert stac_item.to_dict() == reference
```

Comment:
I think we should keep the explicit validation with expected contents. This way, we make sure that the methods did not impact other parts of the properties/metadata shuffling.

Reply:
But it makes it a bit more difficult to test a wider variety of files, i.e. we'll need a reference for each one of those, and we'll have to keep our hard-coded tests in sync with the version found on live servers.

Reply:
I think we should have a set of `{test-ncml-file}.xml` / `{expected-stac-item}.json`, as simply load corresponding paths with `@pytest.mark.parametrize`. 

With local references, things shouldn't drastically change unless we ourselves modify the implementation.

Reply:
It's going to be a bit messier, since we need the catalog.xml and the netCDF xml, but yes, I can work on that.

Reply:
Another problem is that the catalog.xml does not include the host name. So the logic to reconstruct access_urls would have to be different when reading from an xml file on disk and when making a server request, which defeats the purpose of unit tests.

Reply:
The @serviceType varies across servers (e.g. OPENDAP, OpenDAP). Maybe better to use the @name (odap).

Reply:
> Another problem is that the catalog.xml does not include the host name. So the logic to reconstruct access_urls would have to be different when reading from an xml file on disk and when making a server request, which defeats the purpose of unit tests.

I think instead you can use `responses` to mock requests.
So you pass in the URL as if it was the real instance, but with swaps the response with the XML on disk. 

> The `@serviceType` varies across servers (e.g. OPENDAP, OpenDAP). Maybe better to use the `@name` (odap).

I think we should use the standard names (in schema `ServiceTypeEnum`).
However, we might need some kind of custom handling because even the docs seems to provide multiple casing variations: 
- https://docs.unidata.ucar.edu/tds/current/userguide/services_ref.html#standard-data-services
- https://docs.unidata.ucar.edu/tds/current/userguide/services_ref.html#tds-requirements-for-thredds-catalog-service-elements
- https://docs.unidata.ucar.edu/tds/current/userguide/tds_config_ref.html#opendap-service
- https://schemas.unidata.ucar.edu/thredds/InvCatalog.1.2.xsd (see `ServiceTypeEnum`)

I see at least `OPENDAP`, `OPeNDAP`, `opendap`, `Opendap`, and `OpenDAP`.

Reply:
conftest.py now has a function to record the server's response. It's stored in the test directory. 
Another function used the stored response to create STAC item dicts, which can then be used by unit tests to compare results from live servers with a reference.

Added Enum with the serviceTypes defined in the latest version of the schema (1.0.7).